<launch>
  <arg name="launch_teleop" default="true" type="bool"/>
  <!-- left hand -->
  <node ns="sandia_hand_left" pkg="hrpsys_gazebo_atlas" type="hand_controller" name="controller_manager" output="screen" >
    <param name="robot_description" textfile="$(find hrpsys_gazebo_atlas)/models/sandia_hand_left.urdf" />
    <remap from="commands" to="/sandia_hands/l_hand/joint_commands" />
    <remap from="in_joint_states" to="/atlas/joint_states" />
    <!-- <remap from="in_joint_states" to="/sandia_hands/l_hand/joint_states" /> -->
  </node>
  <node ns="sandia_hand_left" pkg="pr2_controller_manager" type="spawner" name="spawner" args="sandia_hand_left_controller" output="screen" >
  </node>
  <rosparam ns="sandia_hand_left"
            file="$(find hrpsys_gazebo_atlas)/config/sandia_hand_left_controller.yaml" command="load" />

  <!-- right hand -->
  <node ns="sandia_hand_right"
        pkg="hrpsys_gazebo_atlas" type="hand_controller" name="controller_manager"
        output="screen" >
    <param name="robot_description" textfile="$(find hrpsys_gazebo_atlas)/models/sandia_hand_right.urdf" />
    <remap from="commands" to="/sandia_hands/r_hand/joint_commands" />
    <remap from="in_joint_states" to="/atlas/joint_states" />
    <!-- <remap from="in_joint_states" to="/sandia_hands/r_hand/joint_states" /> -->
  </node>
  <node ns="sandia_hand_right"
        pkg="pr2_controller_manager" type="spawner" name="spawner"
        args="sandia_hand_right_controller" output="screen" >
  </node>
  <rosparam ns="sandia_hand_right"
            file="$(find hrpsys_gazebo_atlas)/config/sandia_hand_right_controller.yaml" command="load" />
  
  <group if="$(arg launch_teleop)">
    <node ns="sandia_hands/l_hand" name="simple_grasp_left" pkg="sandia_hand_teleop" type="simple_grasp.py"/>
    <node ns="sandia_hands/r_hand" name="simple_grasp_right" pkg="sandia_hand_teleop" type="simple_grasp.py"/>
  </group>
    
</launch>
